electrum/wallet.py

```diff
@@ -3342,6 +3342,12 @@ def restore_wallet_from_text(text, *, path, config: SimpleConfig,
     assert not storage.file_exists(), "file was created too soon! plaintext keys might have been written to disk"
     wallet.update_password(old_pw=None, new_pw=password, encrypt_storage=encrypt_file)
     wallet.synchronize()
+    if wait_for_sync:
+        if not sync_network:
+            raise Exception('sync_network is required if wait_for_sync is True')
+        wallet.start_network(sync_network)
+        while not wallet.network.is_connected:
```

Comment:
`wallet.network.is_connected` has very little to do with the wallet being synchronized. It is ~sort of a necessary but not sufficient condition. You can check `wallet.is_up_to_date()` instead.